### PR TITLE
4G Heating district template

### PR DESCRIPTION
**Pull request by vtnate**
#### Any background context you want to provide?
Bringing in HeatingIndirect.mot
#### What does this PR accomplish?
Allows use of indirect heating template in district system
#### How should this be manually tested?
`py.test tests/model_connectors/test_district_heating_system.py` for the newly added functionality. `py.test` to ensure everything works, because we had to touch a bunch of models in this PR.
#### What are the relevant tickets? #165

### CODE REVIEW

File: geojson_modelica_translator/model_connectors/templates/DistrictHeatingSystem.mot [hunk 1]
```diff
@@ -0,0 +1,129 @@
+    within Buildings.Applications.DHC.CentralPlants.Heating.Generation4.Examples;model DistrictHeatingSystem3BuiNewBoiler
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/templates/DistrictHeatingSystem.mot:1]: This should be `within geojson_modelica_translator.model_connectors.templates;`

Comment (nllong) [reply geojson_modelica_translator/model_connectors/templates/DistrictHeatingSystem.mot:1]: Also, can you make sure to run the modelica-fmt. You just need to install modelica-fmt, then the pre-commit will use it. Grab latest release from the assets here: https://github.com/urbanopt/modelica-fmt/releases/tag/v0.1-pr.3. Make sure to place it in your path.

File: geojson_modelica_translator/model_connectors/district_system.py [hunk 2]
```diff
@@ -209,17 +218,38 @@ def to_modelica(self, scaffold, model_connector_base):
...
+            # previous paths and replace with the new one.
+            # Make sure to update the names of any resources as well.
+            mofile.set_within_statement(f'{scaffold.project_name}.Districts')
+            mofile.save()
 
             self.copy_required_mo_files(
                 scaffold.districts_path.files_dir, within=f'{scaffold.project_name}.Districts'
             )
 
+            if "Heating" in ets_model_type:
```
Comment (nllong) [inline geojson_modelica_translator/model_connectors/district_system.py:234]: nice... 

Yeah, I think we can remove the model_name object here unless the HeatingINdirect.mot is looking for a model_name.

File: geojson_modelica_translator/system_parameters/schema.json [hunk 3]
```diff
@@ -287,8 +287,8 @@
       },
       "additionalProperties": false
     },
-    "ets_indirect_cooling_parameters": {
-      "description": "Indirect cooling ETS. The energy transfer is implemented through nondirect contact of water loops. Parameters associated with indirect cooling ETS",
+    "ets_indirect_parameters": {
```
Comment (nllong) [inline geojson_modelica_translator/system_parameters/schema.json:290]: thanks for updating the schema. Note that this is going to conflict with Amy's PR (https://github.com/urbanopt/geojson-modelica-translator/pull/172) and with https://github.com/urbanopt/geojson-modelica-translator/pull/180. But I guess the first in wins.

File: tests/geojson_modelica_translator/data/system_parameters_mix_models.json [hunk 4]
```diff
@@ -2,18 +2,20 @@
...
           "nominal_flow_building": 0.5,
           "pressure_drop_valve": 1000,
           "pressure_drop_hx_secondary": 1500,
           "pressure_drop_hx_primary": 1500,
-          "supply_water_temperature_district": 5,
-          "supply_water_temperature_building": 7
+          "cooling_supply_water_temperature_district": 5,
+          "cooling_supply_water_temperature_building": 7,
+          "heating_supply_water_temperature_district": 55,
+          "heating_supply_water_temperature_building": 50
```
Comment (nllong) [inline tests/geojson_modelica_translator/data/system_parameters_mix_models.json:18]: yeah, i like this pattern.

File: tests/model_connectors/test_district_system.py [hunk 5]
```diff
@@ -70,18 +70,34 @@ def setUp(self):
...
         )
         run_path = Path(os.path.abspath(self.gj.scaffold.project_path)).parent
-        exitcode = mr.run_in_docker(file_to_run, run_path=run_path, project_name=self.gj.scaffold.project_name)
+        exitcode = mr.run_in_docker(cooling_file_to_run, run_path=run_path, project_name=self.gj.scaffold.project_name)
         self.assertEqual(0, exitcode)
 
         results_path = os.path.join(run_path, f"{self.gj.scaffold.project_name}_results")
         self.assertTrue(os.path.join(results_path, 'stdout.log'))
+
+    # def test_district_heating_to_modelica_and_run(self):
```
Comment (nllong) [inline tests/model_connectors/test_district_system.py:89]: just delete this code.

**Review by nllong — CHANGES_REQUESTED** [review]
small changes requested. Yes this looks good. 

The way to test this is to create a new test file. I'd copy this one (https://github.com/urbanopt/geojson-modelica-translator/blob/develop/tests/model_connectors/test_district_system.py) and call it test_district_heating.py or something like that. Then you will need to update the system parameter file to use indirect with heating.

**Comment by vtnate** [comment]
> small changes requested. Yes this looks good.
> 
> The way to test this is to create a new test file. I'd copy this one (https://github.com/urbanopt/geojson-modelica-translator/blob/develop/tests/model_connectors/test_district_system.py) and call it test_district_heating.py or something like that. Then you will need to update the system parameter file to use indirect with heating.

@nllong It looks like we don't have anything that explicitly tests `CoolingIndirect.mot` and it just gets used during operation of `DistrictCoolingSystem.mot`, is that correct? So therefore the same goes for `HeatingIndirect.mot`?

Something still isn't right with the template, so @Helarga and I are going over it tomorrow morning.

**Comment by vtnate** [comment]
Ugh. @nllong Travis wants modelica-fmt to make changes, but it passed locally before I pushed. Also, Travis canceled the CI build because the district cooling test took longer than 10 minutes. That test takes 17 minutes on my laptop. Can you confirm locally that the tests pass, and maybe adjust the Travis settings?

**Comment by nllong** [comment]
1) Make sure to install modelica fmt 0.1-pr3 (https://github.com/urbanopt/modelica-fmt/releases/tag/v0.1-pr.3). I ran locally and it did make the changes.
2) Yes, the >10 minute is an issue because when running in jmodelica then isn't any data being written out, therefore, travis kills the process. We can either a) figure out how to make the simulation run in < 10 minutes, or b) find a way to echo something to stdout while waiting on jmodelica.  I vote for b.

**Comment by Helarga** [comment]
> 1. Make sure to install modelica fmt 0.1-pr3 (https://github.com/urbanopt/modelica-fmt/releases/tag/v0.1-pr.3). I ran locally and it did make the changes.
> 2. Yes, the >10 minute is an issue because when running in jmodelica then isn't any data being written out, therefore, travis kills the process. We can either a) figure out how to make the simulation run in < 10 minutes, or b) find a way to echo something to stdout while waiting on jmodelica.  I vote for b.
@nllong for the cooling district plant we have used 8 spawn buildings, i will replace it with 2 as the Heating District Plant

**Comment by nllong** [comment]
> > 1. Make sure to install modelica fmt 0.1-pr3 (https://github.com/urbanopt/modelica-fmt/releases/tag/v0.1-pr.3). I ran locally and it did make the changes.
> > 2. Yes, the >10 minute is an issue because when running in jmodelica then isn't any data being written out, therefore, travis kills the process. We can either a) figure out how to make the simulation run in < 10 minutes, or b) find a way to echo something to stdout while waiting on jmodelica.  I vote for b.
> >    @nllong for the cooling district plant we have used 8 spawn buildings, i will replace it with 2 as the Heating District Plant

that will help, thanks @Helarga !

**Comment by Helarga** [comment]
@vtnate  Hi Nate, so locally the simulation on Jmodelica took 29.8 seconds
image

**Comment by vtnate** [comment]
Thanks Hagar!

And sorry Nick, I wasn't clear in my comment. I have modelica-fmt 0.1-pr3 which ran and updated the modelica files locally before I pushed. That's why I was surprised it it failed CI.

**Comment by vtnate** [comment]
Ugh: https://travis-ci.org/github/urbanopt/geojson-modelica-translator/jobs/730353040 modelica-fmt again. All the py.tests passed though!

**Comment by vtnate** [comment]
Welp, I learned my lesson. Need to run all files through pre-commit, not just let the hook run when you make a commit. Hopefully this note can help others avoid this frustration.

**Comment by nllong** [comment]
Ah, right, I should have said this more clearly too. By default pre-commit only runs on files that have changed in the staged commit. If you haven't been keeping up to date, then you need to run `pre-commit run --all-files`

**Review by nllong — APPROVED** [review]
amazing, thanks everyone!

**Pull request merged**